crates/cairo-lint-core/src/lints/manual/helpers.rs

```diff
@@ -116,3 +154,45 @@ pub fn get_else_expr_block(else_clause: OptionElseClause, db: &dyn SyntaxGroup)
         },
     }
 }
+
+/// Checks if the input `Expr` is a default of the expr kind.
+///
+/// # Arguments
+/// * `db` - Reference to the `SyntaxGroup` for syntax tree access.
+/// * `expr` - The target expr.
+///
+/// # Returns
+/// * `true` if the expression is a default otherwise `false`.
+pub fn check_is_default(db: &dyn SyntaxGroup, expr: &Expr) -> bool {
+    match expr {
+        Expr::FunctionCall(func_call) => {
+            let func_name = func_call.path(db).as_syntax_node().get_text_without_trivia(db);
+            func_name == "Default::default" || func_name == "ArrayTrait::new"
+        }
+        Expr::False(expr_false) => !expr_false.boolean_value(),
+        Expr::String(expr_str) => {
+            if let Some(str) = expr_str.string_value(db) {
+                str.is_empty()
+            } else {
+                false
+            }
+        }
+        Expr::Block(expr_block) => {
+            let mut statements_check = false;
+            for statement in expr_block.statements(db).elements(db) {
+                statements_check = match statement {
+                    Statement::Expr(statement_expr) => check_is_default(db, &statement_expr.expr(db)),
+                    _ => false,
+                }
+            }
```

Comment:
pretty scared by this loop, Why should it be a loop ?

Reply:
Haha just removed it.